### DISCUSSION TITLE
Show pictures in Marriage event boxes

### DISCUSSION THREAD

**Discussion by dvpfig** [PeWu/topola-viewer#110] [Q&A] [answered]
Hi, 

I'm using Topola viewer with the Gramps plugin Interactive family tree. 

It's working great and showing the family tree really nice with pictures of persons. 

I'm wondering if there is a way to show a photo of the marriages in the marriage box generated in the Topola view. 

thanks.

**Comment by czifumasa** [comment, marked as answer]
Hi!

Currently it's not possible, but I'm thinking of contributing something like this anyway. Do you mean marriage box directly on the family tree?
![image](https://user-images.githubusercontent.com/15907129/188436199-152da18f-f04a-47de-8c29-8fba9af9a717.png)

or in the details panel?
![image](https://user-images.githubusercontent.com/15907129/188436397-b152bd3c-a0b4-4d65-9f8c-b48859d86bd8.png)

**Comment by dvpfig** [comment]
thanks for the answer. 

First priority is for display purposes, so it's your first option: show picture directly on the family tree. 
It would really cool to have that. 

But, of course, (similar as for photos of persons) the complete way would be to add the first photo on the family tree box and all available photos under the details panel. Maybe that's asking too much, though.

**Comment by czifumasa** [comment]
Both options are doable, but for the first one, we would also need changes in https://github.com/PeWu/topola, so it's a bit more complicated.

One more question:, in Gramps, where do you attach your marriage photos? To marriage event? Or directly to the family?

**Reply by dvpfig** [reply]
I attach the wedding photo to the event marriage. I think the photos in the gallery of the Family should be shown as family photos, rather than under the Marriage event section or box.

**Reply by czifumasa** [reply]
Yeah, that's why I am asking. I am doing it exactly the same as you, but I think many people would like to see photo from family gallery with all children, instead of wedding photo specifically.

Maybe we should just take whatever is available? Or make it configurable?

Before I start contribution, I would like to hear opinion from @PeWu as he is the owner of the project.

**Reply by dvpfig** [reply]
That's a really good point. 

I've thinking about this in terms of showing a photo of a specific date (the marriage) for this family.
But of course there are families in my tree of not married persons. 

I actually change my opinion to the following:
* show a photo/pic of a family in the family box in family tree, and/or in family details in side bar. 
* the photo should be the first photo of gallery in family. 
* the family box may or may not include the date of marriage. 

To my use case of showing a wedding photo in married families, I would just place that photo as the first photo in the family gallery. For non-married families, I may just put a photo of 2 persons or of a couple with kids or whatever it fits best. 

Excellent discussion, btw. 
Thanks

**Comment by PeWu** [comment]
Hi,

I don't have a strong opinion as I don't attach family photos to my genealogy database. Nonetheless, this seems like a great idea.

To implement this feature, 2 repositories need to be changed:
1. [topola](https://github.com/PeWu/topola)
    - Add family image to [API](https://github.com/PeWu/topola/blob/master/src/data.ts#L54)
    - Display image if provided
2. [topola-viewer](https://github.com/PeWu/topola-viewer)
    - Extract family image and pass it to the rendering library
    - Show image in details panel (this can be done independently from displaying the image in the tree)
 
Thanks for the suggestion!
